Mathlib/Topology/Separation.lean

```diff
@@ -1068,6 +1068,17 @@ theorem SeparationQuotient.t1Space_iff : T1Space (SeparationQuotient X) ↔ R0Sp
     erw [mk_eq_mk, inseparable_iff_specializes_and]
     exact ⟨xspecy, yspecx⟩
 
+lemma Set.Subsingleton.isClosed [T1Space X] {A : Set X} (h : A.Subsingleton) : IsClosed A := by
+  rcases h.eq_empty_or_singleton with rfl | ⟨x, rfl⟩
+  · exact isClosed_empty
+  · exact isClosed_singleton
+
+lemma IsClosed.inter_singleton [T1Space X] {A : Set X} {a : X} : IsClosed (A ∩ {a}) :=
```

Comment:
Upon second thought, I think `isClosed_inter_singleton` would be a better name.
*If* this lemma took a hypothesis `IsClosed A`, `IsClosed.inter_singleton` would be a great name and allow access to dot notation. That is not the case here. Same comment applies below.